backend/aim/metrics/m9/m9_umsi.py

```diff
@@ -441,15 +483,21 @@ def execute_metric(
                 original_images, heatmap_batch, heatmap_overlay_batch
             )
 
+        # Concatenate all batch cropped images
```

Comment:
We have concatenated all of the cropped images here. Since the code itself is designed to work with multiple images, we use the multiple image feature. When you are using a GPU, this feature allows for more speed.